### PR TITLE
Route MoE router TopK through the kernel runtime

### PR DESCRIPTION
**Pull request by vikhyat**
## Summary

- Replace the app-side `torch.topk` + `F.softmax` MoE router path with `_KERNELS.moe.topk_fwd`.
- Cache the runtime callable at module import, matching the existing layernorm runtime pattern.

## Dependency

Depends on m87-labs/kestrel-proprietary#93, which exposes `moe.topk_fwd` through the stable kernel runtime and owns fallback behavior there.

## Validation

- `python -m py_compile kestrel/moondream/layers.py scripts/chartqa_eval.py`
- L4 ChartQA comparisons were run through Modal. The TopK microbench is faster, but the paired end-to-end ChartQA runs were noise-level after accounting for order and remaining MoE JIT artifacts:
  - cold-first: patched `2.36 RPS`, baseline `2.49 RPS`
  - warm-second: patched `6.69 RPS`, baseline `6.61 RPS`

### CODE REVIEW

**Comment by vikhyat** [comment]
@codex review

**Comment by vikhyat** [comment]
@codex review
